tests/foreman/api/test_syncplan.py

```diff
@@ -493,43 +494,102 @@ def test_positive_repeatedly_add_remove(self):
             self.assertEqual(len(syncplan.read().product), 0)
 
 
-@stubbed()
 class SyncPlanSynchronizeTestCase(APITestCase):
     """Tests specific to synchronizing sync plans."""
 
     @classmethod
     def setUpClass(cls):
-        """Create an organization and products which can be re-used in
-        tests."""
+        """Create an organization which can be re-used in tests."""
         super(SyncPlanSynchronizeTestCase, cls).setUpClass()
         cls.org = entities.Organization().create()
-        cls.products = [
-            entities.Product(organization=cls.org).create() for _ in range(2)
-        ]
 
-    @stubbed()
+    def validate_repo_content(
+            self, repo, content_types, after_sync=True, max_attempts=10):
+        """Check whether corresponding content is present in repository before
+        or after synchronization is performed
+
+        :param repo: Repository entity instance to be validated
+        :param content_types: List of repository content entities that
+            should be validated (e.g. package, erratum, puppet_module)
+        :param bool after_sync: Specify whether you perform validation before
+            synchronization procedure is happened or after
+        :param int max_attempts: That value is basically introduced for slow
+            systems when user sure that sync procedure can take more than 300
+            seconds (30*10 where 10 is a default value for max_attempts
+            variable)
+
+        """
+        for i in range(max_attempts):
+            try:
+                for content in content_types:
+                    if after_sync:
+                        self.assertGreater(repo.content_counts[content], 0)
+                    else:
+                        self.assertFalse(repo.content_counts[content])
+                break
+            except AssertionError:
+                sleep(30)
+                repo = repo.read()
+        else:
+                raise AssertionError(
+                    'Repository contains invalid number of content entities')
+
     @run_only_on('sat')
     @tier2
     def test_positive_synchronize_product(self):
-        """Create a sync plan with one product and sync it.
+        """Create a sync plan with one product and sync it automatically.
 
-        @Assert: A sync plan is created with one product and product can be
-        synchronized.
+        @Assert: Product is synchronized successfully.
 
         @Feature: SyncPlan
         """
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            enabled=True,
+            sync_date=datetime.utcnow(),
```

Comment:
@sthirugn I've just double checked this via WebUI, the behavior is the same. Seems like if the sync plan is already started and you associate a new product with it, then the product will be synchronized immediately, without waiting for the next scheduled date. Is there someone whom we can ask whether that's expected approach or not? Seems like "it's not a bug, it's a feature" situation, considering the behavior is the same for UI/CLI/API :)